sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/SynchronousMessageSubscriber.java

```diff
@@ -5,114 +5,239 @@
 
```

Comment:
Only thing you need are unit tests. The logic in this subscriber is complex and I can see it being hard to debug.